internal/experiment/generation/generation.go

```diff
@@ -95,18 +96,22 @@ func loadApplicationData(app *redskyappsv1alpha1.Application, src string) ([]byt
 	dst := filepath.Join(os.TempDir(), fmt.Sprintf("load-application-data-%x", md5.Sum([]byte(src))))
 	defer os.Remove(dst)
 
-	opts := []getter.ClientOption{
-		func(c *getter.Client) error {
-			// Try to load relative to the application definition itself
-			if path := app.Annotations["config.kubernetes.io/path"]; path != "" {
-				c.Pwd = filepath.Dir(path)
-			}
+	var opts []getter.ClientOption
+
+	// Only set the working directory to directory of the file the app.yaml was loaded from,
+	// we MUST NOT set this to the process working directory or relative paths in the app.yaml
+	// will be dependent on what directory you run the process from. If the path annotation is
+	// not present on the application, we MUST fail to load relative paths.
+	if path := app.Annotations[kioutil.PathAnnotation]; path != "" {
+		opts = append(opts, func(c *getter.Client) error {
+			c.Pwd = filepath.Dir(path)
```

Comment:
I know earlier we set go through the work of setting this annotation to include the trailing app.yaml filename. Any reason we go through that effort instead of just using the directory itself?

Reply:
The value of the path annotation is the path to the file that the YAML was read from, normally we don't set that (it will come from inside Konjure or inside KYAML). The other code is just using `app.yaml` as a dummy for process streams to match that convention.